raft/src/main/java/org/apache/kafka/snapshot/SnapshotReader.java

```diff
@@ -84,7 +84,7 @@ public int lastContainedLogEpoch() {
      */
     public long lastContainedLogTimestamp() {
         if (!lastContainedLogTimestamp.isPresent()) {
-            // nextBatch is expected to be empty
+            // nextBatch is expected to be equal to Optional.empty() so just replace it
```

Comment:
Can we check this and throw an exception if it's not true? I don't think this method is called often so performance should not be an issue.

Reply:
Done.